src/synthesis.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  availableAmount,
+  itemAmount,
+  mySpleenUse,
+  spleenLimit,
+  sweetSynthesis,
+  sweetSynthesisResult,
+} from "kolmafia";
+import { $items, clamp } from "libram";
+import { shuffle } from "./lib";
+
+const whitelist = $items`sugar shotgun, sugar shillelagh, sugar shank, sugar chapeau, sugar shorts, sugar shield, sugar shirt, Fudgie Roll`;
+export default function synthesize(effect: Effect, casts: number): void {
+  const shuffledWhitelist = shuffle([...whitelist]);
+  for (const itemA of shuffledWhitelist) {
+    if (availableAmount(itemA) <= 1) continue;
+    if (casts === 0) return;
+    for (const itemB of shuffledWhitelist) {
+      const minimum = itemA === itemB ? 2 : 1;
+      if (availableAmount(itemB) <= minimum) continue;
+      if (sweetSynthesisResult(itemA, itemB) !== effect) continue;
+      const possibleCasts =
+        itemA === itemB
+          ? Math.floor((itemAmount(itemA) - 1) / 2)
+          : Math.min(itemAmount(itemA), itemAmount(itemB)) - 1;
+      const spleen = Math.max(spleenLimit() - mySpleenUse(), 0);
+      const castsToDo = Math.min(possibleCasts, casts, spleen);
+      if (castsToDo === 0) continue;
+      sweetSynthesis(castsToDo, itemA, itemB);
```

Comment:
```suggestion
      retrieveItem(itemA, castsToDo);
      retrieveItem(itemB, castsToDo);
      sweetSynthesis(castsToDo, itemA, itemB);
```

Reply:
Is this safe for double-synthing a single thing?

Reply:
that... is a good and annoying question.